BUG: Fix ob_shash deprecation warning on Python 3.11

### Describe the issue:

`ob_shash` is deprecated on Python 3.11:
```
numpy/core/src/multiarray/scalarapi.c:709:13: warning: ‘ob_shash’ is deprecated [-Wdeprecated-declarations]
```
This looks unnecessary :thinking:, cached hash value for `obj` should not be computed yet:
> https://github.com/numpy/numpy/blob/28e624d1e78ee46d7040e6a099bda39d7fa60e3d/numpy/core/src/multiarray/scalarapi.c#L709



### Reproduce the code example:

```python
Run tests on Python 3.11.
```


### Error message:

```shell
numpy/core/src/multiarray/scalarapi.c:709:13: warning: ‘ob_shash’ is deprecated [-Wdeprecated-declarations]
```


### NumPy/Python version information:

1.23.0.dev0+1020.g08569fcf0 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]

--- a/numpy/core/src/multiarray/scalarapi.c
+++ b/numpy/core/src/multiarray/scalarapi.c
@@ -706,7 +706,9 @@ PyArray_Scalar(void *data, PyArray_Descr *descr, PyObject *base)
     if (PyTypeNum_ISFLEXIBLE(type_num)) {
         if (type_num == NPY_STRING) {
             destptr = PyBytes_AS_STRING(obj);
-            ((PyBytesObject *)obj)->ob_shash = -1;
+            #if PY_VERSION_HEX < 0x030b00b0
+                ((PyBytesObject *)obj)->ob_shash = -1;
+            #endif
             memcpy(destptr, data, itemsize);
             return obj;
         }

